### PR TITLE
fix: correctly diff new actor version

### LINKED ISSUE #1144: Efficent state tree diffing is broken in `nv18-rc1`
### Describe the bug:
The default actor type in lotus has changed, it now has a new field: https://github.com/filecoin-project/lotus/blob/master/chain/types/actor.go#L33 this causes unmarshalling actors here to fail: https://github.com/filecoin-project/lily/blob/master/chain/datasource/datasource.go#L393-L416 since their encoded bytes are different depending on network version.

### Steps to Reproduce:
Run a walk or watch job

### PR DESCRIPTION
**Pull request by frrist**
- closes #1144

### CODE REVIEW

File: chain/datasource/datasource.go [hunk 1]
```diff
@@ -376,104 +394,104 @@ func GetGenesisActors(ctx context.Context, store adt.Store, genesis *types.TipSe
...
+				ch.Actor = *types.AsActorV5(&act)
+			} else {
+				var act types.Actor
+				buf.Reset(change.After.Raw)
+				err := act.UnmarshalCBOR(buf)
+				buf.Reset(nil)
+				if err != nil {
+					return nil, err
+				}
+				ch.Actor = act
```
Comment (frrist) [inline chain/datasource/datasource.go:487]: This is the actual fix; we address it by mimicking the behaviour of lotus: https://github.com/filecoin-project/lotus/blob/v1.20.0-rc2/chain/state/statetree.go#L631

File: chain/datasource/datasource.go [hunk 2]
```diff
@@ -294,6 +291,43 @@ func ComputeGasOutputs(ctx context.Context, block *types.BlockHeader, message *t
 	return vm.ComputeGasOutputs(receipt.GasUsed, message.GasLimit, block.ParentBaseFee, message.GasFeeCap, message.GasPremium, burn), nil
 }
 
+type StateTreeMeta struct {
+	// Root is the root od Map
```
Comment (birdychang) [inline chain/datasource/datasource.go:295]: `of`

File: chain/datasource/datasource.go [hunk 3]
```diff
@@ -303,41 +337,26 @@ func GetActorStateChanges(ctx context.Context, store adt.Store, current, execute
...
 	if err != nil {
 		return nil, err
 	}
 
-	if newVersion == oldVersion && (newVersion != types.StateTreeVersion0 && newVersion != types.StateTreeVersion1) {
-		changes, err := fastDiff(ctx, store, oldRoot, newRoot)
+	if newTree.Tree.Version() > 1 && oldTree.Tree.Version() > 1 {
+		changes, err := fastDiff(ctx, store, oldTree, newTree)
 		if err == nil {
-			metrics.RecordInc(ctx, metrics.DataSourceActorStateChangesFastDiff)
```
Comment (birdychang) [inline chain/datasource/datasource.go:319]: I think either keep those counts or add diff time histogram metrics.

**Review by birdychang — APPROVED** [review]

**Review by kasteph — APPROVED** [review]

File: chain/datasource/datasource.go (hunk 3 above)
Comment (frrist) [reply chain/datasource/datasource.go:319]: addressed

**Pull request merged**